File: source_funcs/go_transcoder/QUERIES_COUNTS_ARRAY_ELEMENTS_VALUES_GIVEN_RANGE/QUERIES_COUNTS_ARRAY_ELEMENTS_VALUES_GIVEN_RANGE.go
package main

func f_gold(arr []int, n int, x int, y int) int {
	var count int = 0
	for i := int(0); i < n; i++ {
		if arr[i] >= x && arr[i] <= y {
			count++
		}
	}
	return count
}

File: rust_funcs/go_transcoder/QUERIES_COUNTS_ARRAY_ELEMENTS_VALUES_GIVEN_RANGE/QUERIES_COUNTS_ARRAY_ELEMENTS_VALUES_GIVEN_RANGE.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


fn f_gold(arr: [i32;2], n: i32, x: i32, y: i32) -> i32 {
    let mut count = 0;
    for i in 0..n {
        if arr[i as usize] >= x && arr[i as usize] <= y {
            count += 1;
        }
    }
    count
}
